_tutorials/gen-ai/agents/build-plan-execute-reflect-agent.md

```diff
@@ -0,0 +1,522 @@
+    }
+  }
+}
+```
+
+`-1` implies no limit on retries.
+
+### Step 2: Create an agent
+
+Create an agent of `plan_execute_and_reflect` type.
+
+The agent is configured with the following information:
+
+- Meta information: `name`, `type`, `description`.
+- LLM information: The agent uses an LLM to reason, come up with a plan for completing the task, execute the steps with appropriate tools, and reflect on the intermediate results to optimize the plan.
+- Tools: A tool is a function that can be executed by the agent. Each tool can define its own `name`, `description`, `parameters` and `attributes`.
+- Memory: Stores chat messages. Currently, OpenSearch only supports one memory type: `conversation_index`.
+
+The agent contains the following parameters:
+
+- `app_type`: Specify this parameter for reference purposes in order to differentiate between multiple agents.
+- `llm`: Defines the LLM configuration.
+    - `model_id`: Model to be used for this agent.
+    - `parameters`: Additional configuration for the model.
+- `parameters`: Agent parameters. It can be used to control the `max_steps` executed by the agent, modify default prompts, etc.
+    - `llm_interface`: Defines default parameters for the model to be used for function calling and path to the llm response. This is required as each LLM can return the response in it's own format. Currently, OpenSearch supports defaults for 3 models:
+        - `bedrock/converse/claude`: For bedrock converse claude models.
+        - `openai/v1/chat/completions`: For OpenAI chat completion models.
+        - `bedrock/converse/deepseek_r1`: For bedrock converse deepseek models. 
+    Since the `llm_interface` only sets defines defaults. You can setup function calling for any model by populating the internal `parameters`. For more information, please see [Function Calling Defaults](#Function-calling-defaults)
+    - `max_steps`: Control the max number of steps executed by the agent.
+    - `executor_agent_id`: This agent internally uses a `conversational` agent to execute each step. An executor agent is created by default using the same model. However, you can create and pass your own agent via this field. This can be useful if you wish to use different models for planning and execution.
+- `memory`: Defines how to store messages. Currently, OpenSearch only supports the `conversation_index` memory, which stores messages in a memory index.
+- Tools: 
+   - An LLM will reason to decide which tool to run and will prepare the tool's input. 
+   - By default, the tool's `name` is the same as the tool's `type`, and each tool has a default description. You can override the tool's `name` and `description`.
+   - Each tool in the `tools` list must have a unique name. Each tool has a custom description so that the LLM can easily understand what the tool does.
+   - Each tool must be accompanied by an `attributes` field used to define the expected `input_schema` of the tool. 
```

Comment:
Explain that `input_schema` is defined with json schema, add json schema link https://json-schema.org/